kernel/arch/dreamcast/kernel/init.c

```diff
@@ -144,6 +147,8 @@ int  __weak arch_auto_init(void) {
     /* Initialize memory management */
     mm_init();
 
+    KOS_INIT_FLAG_CALL(mmu_init);
```

Comment:
Is there a reason for this other than to save needing the user to call this on their own? It's not like the other subsystems where you'd enable a better ability to garbage collect, since it's never been called by default. And a user would still have to configure the mmu translations they want to use before they could leverage the mode. Though I do get how that would bring the mmu into functioning like the various other subsystems we initialize in `arch_auto_init`

It seems to me like the better use for a compile-time setting for MMU would be to do something like disable it entirely, so that a function like `mmu_enabled()` could be a static `false`.

Reply:
Consistency? IDK, it just makes sense to me.

People might want to try porting stuff that use `mmap()`, having MMU as an init flag means that they wouldn't have to hack the code to get it to work on KOS.

Reply:
It is possible for some C++ or Rust or static initializer/constructor function to call `mmap()` or call some nontrivial function during initialization before `main()` even starts that could potentially rely on the MMU.